### PR TITLE
Minor updates for the Nemo Developer Guide

### PR DESCRIPTION
**Pull request by hkelly33**
I made minor updates to these files:
\sources\source\tutorials\Custommodules.rst 
\sources\source\tutorials\weightsharing.rst
\sources\source\tutorials\complex_training.rst
\sources\source\installation.rst
\sources\source\index.rst
\sources\source\training.rst
\sources\source\asr\tutorial.rst
 for the Nemo Developer Guide displayed here: https://nvidia.github.io/NeMo/index.html

### CODE REVIEW

File: docs/sources/source/training.rst [hunk 1]
```diff
@@ -42,10 +42,11 @@ Use `torch.distributed.launch` package to run your script like this (assuming 8
 Example
 ~~~~~~~
 
-Please refer to the `<nemo_git_repo_root>/examples/asr/jasper.py` for comprehensive example. 
+Please refer to the `<nemo_git_repo_root>/examples/asr/jasper.py` for a comprehensive example. 
 It builds one train DAG and up to three validation DAGs to evaluate on different datasets.
 
-Assuming, you are working with Volta-based DGX, you can run training like this:
+If you are working with a +
```
Comment (blisc) [inline docs/sources/source/training.rst]: Why do we have a '+' sign here?

File: docs/sources/source/tutorials/custommodules.rst [hunk 2]
```diff
@@ -28,7 +28,7 @@ Trainable Module
     Notice that :class:`TrainableNM<nemo.backends.pytorch.nm.TrainableNM>` class
     has two base classes: :class:`NeuralModule<nemo.core.neural_modules.NeuralModule>` class and ``torch.nn.Module``.
 
-Define module from scratch
+Defining a module from scratch
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (blisc) [inline docs/sources/source/tutorials/custommodules.rst]: Can you extend this underline such that it is longer than the preceding text?

File: docs/sources/source/index.rst [hunk 3]
```diff
@@ -1,5 +1,5 @@
-NVIDIA Neural Modules: NeMo
-===========================
+NVIDIA Neural Modules Developer Guide
```
Comment (chiphuyen) [inline docs/sources/source/index.rst]: I'd try to keep NeMo in there, e.g. `NVIDIA NeMo (Neural Modules) Developer Guide`

File: docs/sources/source/index.rst [hunk 4]
```diff
@@ -23,7 +23,7 @@ A "Neural Module" is a block of code that computes a set of outputs from a set o
 
 Neural Modules’ inputs and outputs have Neural Type for semantic checking.
 
-An application built with NeMo application is a Directed Acyclic Graph (DAG) of connected modules enabling researchers to define and build new speech and nlp networks easily through API Compatible modules.
+An application built with a NeMo application is a Directed Acyclic Graph (DAG) of connected modules enabling researchers to define and build new speech and nlp networks easily through API Compatible modules.
```
Comment (chiphuyen) [inline docs/sources/source/index.rst]: `An application built with NeMo is a ...`

File: docs/sources/source/training.rst (hunk 1 above)
Comment (chiphuyen) [reply docs/sources/source/training.rst]: `<nemo_git_repo_root>/examples/asr/jasper.py` doesn't compile for some reason

**Comment by hkelly33** [comment]
The plus sign in part of the Sphinx syntax in the .rst  file that helps produce an effect in the document.

**Comment by hkelly33** [comment]
Nemo is explained in the following sentence.

**Comment by hkelly33** [comment]
What is your question here?

**Comment by hkelly33** [comment]
This line affects the size of the header and is not visible in the doc.

File: docs/sources/source/training.rst (hunk 1 above)
Comment (blisc) [reply docs/sources/source/training.rst]: > The plus sign in part of the Sphinx syntax in the .rst file that helps produce an effect in the document.

Unfortunately, this is what is rendered in the final doc
image
Notice the '+' is there

File: docs/sources/source/tutorials/custommodules.rst (hunk 2 above)
Comment (blisc) [reply docs/sources/source/tutorials/custommodules.rst]: > This line affects the size of the header and is not visible in the doc.

Sphinx requires all underlines to be longer than the preceding title.

File: docs/sources/source/training.rst (hunk 1 above)
Comment (hkelly33) [reply docs/sources/source/training.rst]: Correcting.

File: docs/sources/source/index.rst (hunk 4 above)
Comment (hkelly33) [reply docs/sources/source/index.rst]: Correcting.

File: docs/sources/source/asr/tutorial.rst [hunk 5]
```diff
@@ -2,19 +2,19 @@ Tutorial
 ========
 
 Make sure you have installed ``nemo`` and ``nemo_asr`` collection.
-See :ref:`installation` section.
+See the :ref:`installation` section.
 
 .. note::
-    You only need `nemo` and `nemo_asr` collection for this tutorial.
+    You only need the `nemo` and `nemo_asr` collections for this tutorial.
```
Comment (chiphuyen) [inline docs/sources/source/asr/tutorial.rst]: `nemo_asr` is a collection, `nemo` isn't.

Comment (chiphuyen) [reply docs/sources/source/asr/tutorial.rst]: we should also keep line 4 and 8 consistent

Comment (blisc) [reply docs/sources/source/asr/tutorial.rst]: Let's make this line `installed ``nemo`` and the ``nemo_asr`` collection.`
And make line 8 similar.

**Pull request merged**